src/store/index.js

```diff
@@ -24,12 +37,81 @@ export default new Vuex.Store({
     [mutation.CHANGE_PROJECT_VIDEOS](state, newVIDEOS) {
       state.project.video_urls = newVIDEOS;
     },
+    [mutation.NEW_SENTENCE](state, { element: { sentence, comboIndex }, index }) {
+      state.segments.splice(index, 0, { sentence, comboIndex });
+      selectionOffset(state, index, 1);
+    },
+    [mutation.REMOVE](state, index) {
+      state.segments.splice(index, 1);
+      selectionOffset(state, index, 0);
+    },
+    [mutation.CHANGE_COMBO_INDEX](state, { row, newComboIndex }) {
+      state.segments[row].comboIndex = newComboIndex;
+    },
+    [mutation.ADD_SELECTED](state, newSelected) {
+      state.selected.push(newSelected);
+    },
+    [mutation.REMOVE_SELECTED](state, index) {
+      state.selected.splice(index, 1);
+    },
+    [mutation.RESET_SELECTED](state) {
+      state.selected = [];
+    },
+    [mutation.COPY_SELECTED](state) {
+      state.clipboard = [];
+      state.selected.forEach((element) => state.clipboard.push(element));
+    },
+    [mutation.CHANGE_SENTENCE](state, { index, newSentence }) {
+      state.segments[index].sentence = newSentence;
+    },
   },
   actions: {
-    CREATE_PROJECT(state, newProject) {
-      state.commit('CHANGE_PROJECT_NAME', newProject.name);
-      state.commit('CHANGE_PROJECT_SEED', newProject.seed);
-      state.commit('CHANGE_PROJECT_VIDEOS', newProject.video_urls);
+    [action.CREATE_PROJECT]({ commit }, newProject) {
+      commit(mutation.CHANGE_PROJECT_NAME, newProject.name);
+      commit(mutation.CHANGE_PROJECT_SEED, newProject.seed);
+      commit(mutation.CHANGE_PROJECT_VIDEOS, newProject.video_urls);
+    },
+    [action.command.NEW_EMPTY_SENTENCE](context, index) {
+      context.commit(mutation.NEW_SENTENCE, { element: { sentence: '', comboIndex: 0 }, index });
+    },
+    [action.command.NEW_SENTENCE](context, indexList) {
+      indexList.forEach((index) => {
+        context.commit(mutation.NEW_SENTENCE, {
+          element: {
+            sentence: context.state.segments[index].sentence,
+            comboIndex: context.state.segments[index].comboIndex,
+          },
```

Comment:
```suggestion
          element: {...context.state.segments[index]},
```
En soi tu pourrais faire `element: context.state.segments[index},`, mais tu aurais des soucis car les deux seraient une référence vers le même objet (si tu en modifies l'un, l’autre est modifié aussi)